src/modules/75-cf/components/DetailPageTemplate/DetailPageTemplate.tsx

```diff
@@ -29,49 +30,59 @@ export interface DetailPageTemplateProps {
   /** Optional extra components to be added into header (context menu, edit button, etc...)  */
   /** Note: Use absolute position to style it */
   headerExtras?: React.ReactNode
+  menuItems?: React.ReactNode
```

Comment:
Instead of this being a `ReactNode`, change it to be the items from the RBAC options menu.

```suggestion
  menuItems?: RbacOptionsMenuButtonProps['items']
```

Reply:
I think I may have misunderstood what you meant by having an array of items within `menuItem`. Did you mean each  `item` is one of these options, e.g. `Delete` / `Edit`? 
![image](https://user-images.githubusercontent.com/95680496/161582421-01e058d7-9447-488a-b6f2-e2f77de75339.png)

Reply:
You do not want each page passing elements to pop out there, you want to just allow each page to pass a list of items they want to display in a menu. If the page doesn't need this, they can either pass an empty array, or pass nothing. No page should need to worry about the position of this menu or what component will be displayed.

So, in a page, they'd pass something like this.

```typescript
<DetailPageTemplate
  /* ... */
  menuItems={[
    {
      icon: 'cross',
      text: getString('delete'),
      onClick: deleteTargetGroupDialog,
      permission: {
        resource: {
          resourceType: ResourceType.ENVIRONMENT,
          resourceIdentifier: targetGroup.environment
        },
        permission: PermissionIdentifier.DELETE_FF_TARGETGROUP
      }
    }
  ]}
>
  /* ... */
</DetailPageTemplate>
```

In the template that would translate to...

```typescript
{!!menuItems.length && (
  <RbacOptionsMenuButton
    className={css.optionsMenu}
    items={menuItems}
  />
)}